### PR TITLE
[e2e] Remove unused function and clean up redundant code

### PR DESCRIPTION
**Pull request by jiatongw**
**What type of PR is this?**

 /kind cleanup

**What this PR does / why we need it**:

This is a first step clean up for `e2e/framework/service_util.go`. This is a big file with 1600 lines of code, and many of the exported functions/methods don't have comment. we should consider split this file into small parts. 

Thoughts on 

- split and move all unexported functions/methods into a kind of "helper.go"

- split exported functions/methods into small pieces by their functionality 

**Which issue(s) this PR fixes**:

Part of issue #76206 (parent issue #75601 )

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by jiatongw — COMMENTED** [review]
Please also see :
1. line 889, `func TestReachableHTTP`. I was thinking removing it, and in the callers, just replace with `TestReachableHTTPWithRetriableErrorCodes(host, port, []int{}, timeout)` . But I was afraid that pass a `[]int{}` may result confusion.

2. line 1463 `func CleanupServiceResources` and line 1499 `func EnableAndDisableInternalLB()`. I was also thinking removing them. But from the caller, we have to replace them with `framework.TestContext.CloudConfig.Provider.CleanupServiceResources()` and `framework.TestContext.CloudConfig.Provider.EnableAndDisableInternalLB()`, which is ugly..

**Comment by andrewsykim** [comment]
@jiatongw it seems like there's only 1 caller for `CleanupServiceResources` and `EnableAndDisableInternalLB` so I would prefer to get rid of them and call it from the text context directly. What do you think?

**Comment by jiatongw** [comment]
> @jiatongw it seems like there's only 1 caller for `CleanupServiceResources` and `EnableAndDisableInternalLB` so I would prefer to get rid of them and call it from the text context directly. What do you think?

Sounds good!

File: test/e2e/framework/service_util.go [hunk 1]
```diff
@@ -1106,20 +1099,6 @@ func (t *ServiceTestFixture) BuildServiceSpec() *v1.Service {
 	return service
 }
 
-// CreateWebserverRC creates rc-backed pods with the well-known webserver
-// configuration and records it for cleanup.
-func (t *ServiceTestFixture) CreateWebserverRC(replicas int32) *v1.ReplicationController {
```
Comment (andrewsykim) [inline test/e2e/framework/service_util.go]: Verified https://cs.k8s.io/?q=CreateWebserverRC&i=nope&files=&repos=kubernetes/kubernetes

File: test/e2e/network/service.go [hunk 2]
```diff
@@ -91,7 +91,7 @@ var _ = SIGDescribe("Services", func() {
 		}
 		for _, lb := range serviceLBNames {
 			framework.Logf("cleaning load balancer resource for %s", lb)
-			framework.CleanupServiceResources(cs, lb, framework.TestContext.CloudConfig.Region, framework.TestContext.CloudConfig.Zone)
+			framework.TestContext.CloudConfig.Provider.CleanupServiceResources(cs, lb, framework.TestContext.CloudConfig.Region, framework.TestContext.CloudConfig.Zone)
```
Comment (andrewsykim) [inline test/e2e/network/service.go]: I actually changed my mind 😆. I think you were right the first time, let's keep the framework wrappers for these for now and clean it up in a future PR

Comment (jiatongw) [reply test/e2e/network/service.go]: Got it. And I think we can also move this file into `framework/service` and split it in a future PR

**Comment by andrewsykim** [comment]
/approve
/lgtm

Thanks @jiatongw

**Pull request merged**